Sklep/MainWindow.cs

```diff
@@ -34,17 +50,77 @@ private void BarcodeScanner_CodeScanned(object sender, string code)
             {
                 barCodeTextBox.Text = code;
             }));
+            playCashRegisterBeep();
+            addProductToList(code);
+         
+        }
+        private void addProductToList(string scannedBarcode)
+        {
+            if (scannedBarcode != "")
+            {
+                var query = db.Products.SingleOrDefault(p => p.Barcode == scannedBarcode);
+                if (query != null)
+                {
+                    listOfProducts.Invoke(new MethodInvoker(delegate
+                    {
+                        var position = receiptPositionList.Find(p => p.barcode == scannedBarcode);
+
+                        if (position != null)
+                        {
+                            position.Amount++;
+                        }
+                        else
+                        {
+                            receiptPositionList.Add(new ReceiptPosition()
+                            {
+                                id = receiptPositionList.Count,
+                                barcode = query.Barcode,
+                                ProductLongName = query.LongName,
+                                PricePerUnit = Convert.ToDecimal(query.Price),
+                                Amount = 1,
+                                Location = new System.Drawing.Point(4, 4 + receiptPositionList.Count * 32),
+                                Size = new System.Drawing.Size(440, 24),
+                            });
+                            receiptPositionList.Last().RemoveButtonClick += ReceiptPosition_RemoveButtonClick;
+                            listOfProducts.Controls.Add(receiptPositionList.Last());
+                        }
+                    }));
+                }
+            }
+        }
+
+        private void ReceiptPosition_RemoveButtonClick(object sender, int parentId)
+        {
+            listOfProducts.Controls.Remove(receiptPositionList[parentId]);
+            receiptPositionList.Remove(receiptPositionList[parentId]);
+            for (int i = parentId; i < receiptPositionList.Count; i++)
+            {
+                receiptPositionList[i].id = i;
+                receiptPositionList[i].Location = new System.Drawing.Point(4, 4 + i * 32);
+            }
+        }
+        private void playCashRegisterBeep()
+        {
+            SoundPlayer simpleSound = new SoundPlayer(AudioResources.Cash_register_beep);
+            simpleSound.Play();
         }
 
         private void MainWindow_FormClosing(object sender, FormClosingEventArgs e)
         {
             barcodeScanner.stopScanning();
         }
 
+        private void AddProductButton_Click(object sender, EventArgs e)
```

Comment:
dobrze by było to wywoływać także, gdy w textboxie do wpisywania kodu wciśniemy enter